sri/templatetags/sri.py

```diff
@@ -8,28 +8,42 @@
 from django.utils.safestring import mark_safe
 
 from sri.algorithm import DEFAULT_ALGORITHM, Algorithm
+from sri.extra_attribute import ExtraAttribute
 from sri.integrity import calculate_integrity_of_static
 
 USE_SRI = getattr(settings, "USE_SRI", not settings.DEBUG)
 
 register = template.Library()
 
 
-def sri_js(attrs: dict, path: str, algorithm: Algorithm):
-    attrs.update({"type": "text/javascript", "src": static(path)})
+def sri_js(attrs: dict, path: str, algorithm: Algorithm, **kwargs):
+    if 'extra_attr' in kwargs:
+        extra_attr = kwargs['extra_attr']
+        if ExtraAttribute(extra_attr) in (ExtraAttribute.PRELOAD, ExtraAttribute.PREFETCH):
+            attrs.update({"rel": extra_attr, "href": static(path), "as": "script"})
+        elif ExtraAttribute(extra_attr) in (ExtraAttribute.DEFER, ExtraAttribute.ASYNC):
+            attrs.update({extra_attr: True})
+            attrs.update({"type": "text/javascript", "src": static(path)})
+    else:
+        attrs.update({"type": "text/javascript", "src": static(path)})
     return mark_safe(f"<script{flatatt(attrs)}></script>")
 
 
-def sri_css(attrs: dict, path: str, algorithm: Algorithm):
-    attrs.update({"rel": "stylesheet", "type": "text/css", "href": static(path)})
+def sri_css(attrs: dict, path: str, algorithm: Algorithm, **kwargs):
+    if 'extra_attr' in kwargs:
+        extra_attr = kwargs['extra_attr']
+        if ExtraAttribute(extra_attr) in (ExtraAttribute.PRELOAD, ExtraAttribute.PREFETCH):
+            attrs.update({"rel": extra_attr, "href": static(path), "as": "style"})
```

Comment:
When using these extra `rel`s, you have to add an extra `onload` to ensure the styles get applied correctly, and a `noscript` fallback.

Perhaps this should be considered out-of scope?

Reply:
I thought you would first do a call setting the value of `extra_attr` to `preload`, then one call without `extra_attr`. Thus it doesn't need that complication.

Reply:
That'd work for the `noscript` fallback, but not for the `onload` requirement.

Reply:
I don't see the point of handling `noscript` and `onload` though, as I don't see them required in specifications when using `defer` or `async`.